packages/stores/src/stores/FeatureFlagsStore.ts

```diff
@@ -0,0 +1,30 @@
+import BaseStore from './BaseStore';
+
+const FLAGS = {
+    foo: false,
+    bar: false,
+    baz: false,
+};
+
+export default class FeatureFlagsStore extends BaseStore<{ [k in keyof typeof FLAGS]: boolean }> {
+    constructor() {
+        super('FeatureFlagsStore', () => {
+            // Set the default values for the first time.
+            if (!this.data) this.update(FLAGS);
+
+            // Update the store data if a new flag was added or removed.
+            if (this.data && Object.keys(this.data).length !== Object.keys(FLAGS).length) {
+                this.update(old => {
+                    const data = FLAGS;
+
+                    Object.keys(FLAGS).forEach(flag => {
+                        // @ts-expect-error flag key is always present in the object, Hence can ignore the TS error.
+                        if (old[flag] !== undefined) data[flag] = old[flag];
+                    });
+
+                    return data;
+                });
+            }
+        });
```

Comment:
why did you add the second arg to BaseStore's constructor?

Reply:
@shayan-deriv Since we are persisting the store's data using `mobx-persist-store` and this process is asynchronous we need to wait for the read to finish and then update the store values, Otherwise, it will always change back to the data in the local storage since that happens after.